### PR TITLE
Fix Background Merge Breaking Snapshot Restore Test

### LINKED ISSUE #63476: [CI] DedicatedClusterSnapshotRestoreIT.testRestoreIndexWithShardsMissingInLocalGateway failure
**Build scan**:
https://gradle-enterprise.elastic.co/s/jeyrxd3tacn7c

**Repro line**:
./gradlew ':server:internalClusterTest' --tests "org.elasticsearch.snapshots.DedicatedClusterSnapshotRestoreIT.testRestoreIndexWithShardsMissingInLocalGateway" \
  -Dtests.seed=65E525FA968063DA \
  -Dtests.security.manager=true \
  -Dtests.locale=hr \
  -Dtests.timezone=Etc/GMT-2 \
  -Druntime.java=8

**Reproduces locally?**:
no

**Applicable branches**:
7.x

**Failure history**:
seen nothing recent

**Failure excerpt**:
java.lang.AssertionError: 
Expected: a value equal to or greater than <3>
     but: <2> was less than <3>
	at __randomizedtesting.SeedInfo.seed([65E525FA968063DA:92EF1F5C86019C56]:0)
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
	at org.junit.Assert.assertThat(Assert.java:956)
	at org.junit.Assert.assertThat(Assert.java:923)
	at org.elasticsearch.snapshots.DedicatedClusterSnapshotRestoreIT.testRestoreIndexWithShardsMissingInLocalGateway(DedicatedClusterSnapshotRestoreIT.java:617)

### PR DESCRIPTION
**Pull request by original-brownbear**
If we run into a background merge between creating the snapshot and closing the index
then with compound files we could be in a situation where we get zero file reuse
on restore.
Force merging before the snapshot gives us a single segment that won't change down the line
so the restore always sees file reuse from the close index.

Closes #63476

### CODE REVIEW

File: server/src/internalClusterTest/java/org/elasticsearch/snapshots/DedicatedClusterSnapshotRestoreIT.java [hunk 1]
```diff
@@ -562,11 +561,8 @@ public void testRestoreIndexWithMissingShards() throws Exception {
 
     public void testRestoreIndexWithShardsMissingInLocalGateway() throws Exception {
         logger.info("--> start 2 nodes");
-        Settings nodeSettings = Settings.builder()
```
Comment (original-brownbear) [inline server/src/internalClusterTest/java/org/elasticsearch/snapshots/DedicatedClusterSnapshotRestoreIT.java:565]: These are unnecessary, we already set this rebalance to none in the parent (`AbstractSnapshotIntegTestCase`)

**Comment by original-brownbear** [comment]
@dnhatn adding you as the reviewer here since we already had the same issue in the index stats ITs a few times. Hope that's ok :)

**Review by dnhatn — APPROVED** [review]
LGTM.

**Comment by original-brownbear** [comment]
Thanks Nhat!

**Pull request merged**